### PR TITLE
feat(DWS): upgrade API of dws cluster

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:
1. The create API has changed that the field tags is removed, use new tag API, instead
2. upgrade the detail API, add checkdeleted

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
 make testacc TEST='./huaweicloud/services/acceptance/dws' TESTARGS='-run=TestAccResourceDWS_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dws -v -run=TestAccResourceDWS_basic -timeout 360m -parallel 4
=== RUN   TestAccResourceDWS_basic
=== PAUSE TestAccResourceDWS_basic
=== CONT  TestAccResourceDWS_basic
--- PASS: TestAccResourceDWS_basic (2681.19s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dws       2681.235s
```

### CODE REVIEW

File: docs/resources/dws_cluster.md [hunk 1]
```diff
@@ -47,48 +47,56 @@ The following arguments are supported:
...
 
 The `public_ip` block supports:
 
 * `eip_id` - (Optional, String, ForceNew) EIP ID.
+  Changing this creates a new cluster resource.
 
 * `public_bind_type` - (Optional, String, ForceNew) Binding type of an EIP. The value can be either of the following:
-  auto_assign not_use bind_existing The default value is not_use.
+  **auto_assign**, **not_use**, **bind_existing**. The default value is **not_use**.
+  Changing this creates a new cluster resource.
```
Comment (chengxiangdong) [inline docs/resources/dws_cluster.md:97]: > The default value is **not_use**.

I didn't find where to set the default value, and it's also required in the API.

File: docs/resources/dws_cluster.md [hunk 2]
```diff
@@ -47,48 +47,56 @@ The following arguments are supported:
...
-* `enterprise_project_id` - (Optional, String, ForceNew) Specifies the enterprise project id of the dws cluster,
+* `enterprise_project_id` - (Optional, String, ForceNew) The enterprise project id of the dws cluster,
   Value 0 indicates the default enterprise project.
-  Changing this parameter will create a new resource.
+  Changing this creates a new cluster resource.
 
 * `availability_zone` - (Optional, String, ForceNew) AZ in a cluster.
+  Changing this creates a new cluster resource.
 
 * `port` - (Optional, Int) Service port of a cluster (8000 to 10000). The default value is 8000.
```
Comment (chengxiangdong) [inline docs/resources/dws_cluster.md]:
```suggestion
* `port` - (Optional, Int, ForceNew) Service port of a cluster (8000 to 10000). The default value is 8000.
  Changing this creates a new cluster resource.
```

File: docs/resources/dws_cluster.md [hunk 3]
```diff
@@ -47,48 +47,56 @@ The following arguments are supported:
...
+  Changing this creates a new cluster resource.
 
 * `public_bind_type` - (Optional, String, ForceNew) Binding type of an EIP. The value can be either of the following:
-  auto_assign not_use bind_existing The default value is not_use.
+  **auto_assign**, **not_use**, **bind_existing**. The default value is **not_use**.
+  Changing this creates a new cluster resource.
 
-* `number_of_cn` - (Optional, int, ForceNew) Specifies the number of CN. If you use a large-scale cluster, deploy
-  multiple CNs.
+* `number_of_cn` - (Optional, int) The number of CN. If you use a large-scale cluster, deploy multiple CNs.
```
Comment (chengxiangdong) [inline docs/resources/dws_cluster.md]:
```suggestion
* `number_of_cn` - (Optional, int, ForceNew) The number of CN. If you use a large-scale cluster, deploy multiple CNs. 
  Defaults to 3.
  Changing this creates a new cluster resource.
```
The default value on the API is 3, and it is up to you whether to add description.

**Comment by chengxiangdong** [comment]
The `PollInterval` value seems to be a wrong value, I think it should be `20 * time.Second`.

```go
func checkClusterDeleteResult(ctx context.Context, client *golangsdk.ServiceClient, clusterId string,
	timeout time.Duration) error {
	stateConf := &resource.StateChangeConf{
		Pending: []string{"Pending"},
		Target:  []string{"Done"},
		Refresh: func() (interface{}, string, error) {
			_, err := cluster.Get(client, clusterId)
			if err != nil {
				if _, ok := parseDwsClusterNotFoundError(err).(golangsdk.ErrDefault404); ok {
					return true, "Done", nil
				}
				return nil, "", nil
			}
			return true, "Pending", nil
		},
		Timeout:      timeout,
		PollInterval: 20 * timeout,  << should be: 20 * time.Second,
		Delay:        20 * time.Second,
	}
	_, err := stateConf.WaitForStateContext(ctx)
	if err != nil {
		return fmt.Errorf("error waiting for DWS (%s) to be delete: %s", clusterId, err)
	}
	return nil
}
```

The same issues exists in other functions as well:

`checkClusterCreateResult`
`checkAndWaitClusterStateAvailable`

File: huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go [hunk 4]
```diff
@@ -491,7 +590,7 @@ func checkClusterDeleteResult(ctx context.Context, client *golangsdk.ServiceClie
 		Refresh: func() (interface{}, string, error) {
 			_, err := cluster.Get(client, clusterId)
 			if err != nil {
-				if _, ok := err.(golangsdk.ErrDefault404); ok {
+				if _, ok := parseDwsClusterNotFoundError(err).(golangsdk.ErrDefault404); ok {
 					return true, "Done", nil
 				}
 				return nil, "", nil
```
Comment (chengxiangdong) [inline huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go]:
```suggestion
				return nil, "failed", err
```

File: docs/resources/dws_cluster.md [hunk 5]
```diff
@@ -47,48 +47,56 @@ The following arguments are supported:
   consist of letters, digits, hyphens(-), or underscores(_) only and must start with a letter.
```
Comment (chengxiangdong) [inline docs/resources/dws_cluster.md:47]:
```suggestion
  consist of letters, digits, hyphens(-), or underscores(_) only and must start with a letter.
  Changing this creates a new cluster resource.
```

File: docs/resources/dws_cluster.md (hunk 1 above)
Comment (hmbig2) [reply docs/resources/dws_cluster.md:97]: It's the default value of API. 
public_ip.public_bind_type, is sub-attr, not required

File: docs/resources/dws_cluster.md (hunk 2 above)
Comment (hmbig2) [reply docs/resources/dws_cluster.md]: done

File: docs/resources/dws_cluster.md (hunk 3 above)
Comment (hmbig2) [reply docs/resources/dws_cluster.md]: done

File: docs/resources/dws_cluster.md (hunk 5 above)
Comment (hmbig2) [reply docs/resources/dws_cluster.md:47]: done

File: huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go (hunk 4 above)
Comment (hmbig2) [reply huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go]: done

File: docs/resources/dws_cluster.md (hunk 1 above)
Comment (chengxiangdong) [reply docs/resources/dws_cluster.md:97]: Okay, got it

**Pull request merged**